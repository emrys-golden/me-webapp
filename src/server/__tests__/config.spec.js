import config from '../config';

jest.unmock('express');

describe('router config tests', () => {
  it('should be defined correctly', () => {
    expect(config).toMatchSnapshot();
  });

  config.forEach((route) => {
    describe(route.path, () => {
      it('should contain a valid path', () => {
        expect(route.path).toBeDefined();
        expect(route.path.charAt(0)).toBe('/');
        expect(route.path[route.path.length - 1]).not.toBe('/');
      });

      it('should contain a valid router', () => {
        expect(route.router).toBeDefined();
        expect(route.router).toBeInstanceOf(Function);
      });
    });
  });
});
