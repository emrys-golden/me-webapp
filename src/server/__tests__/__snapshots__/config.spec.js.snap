// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`router config tests should be defined correctly 1`] = `Array []`;
